test/fft.jl

```diff
@@ -64,153 +27,299 @@ true_fftd2_m4 = [
    34.    3 +  5im   -4    3 -  5im ;
    34.  -11 - 13im    4  -11 + 13im ]
 
-for i = 1:length(m4)
-    @test_approx_eq fft_m4[i] true_fft_m4[i]
-    @test_approx_eq fftd2_m4[i] true_fftd2_m4[i]
-    @test_approx_eq ifft_fft_m4[i] m4[i]
-    @test_approx_eq fftn_m4[i] true_fftn_m4[i]
-    @test_approx_eq ifftn_fftn_m4[i] m4[i]
-
-    @test_approx_eq fft!_m4[i] true_fft_m4[i]
-    @test_approx_eq fft!d2_m4[i] true_fftd2_m4[i]
-    @test_approx_eq ifft!_fft_m4[i] m4[i]
-    @test_approx_eq fft!n_m4[i] true_fftn_m4[i]
-    @test_approx_eq ifft!n_fftn_m4[i] m4[i]
-
-    @test_approx_eq pfft_m4[i] true_fft_m4[i]
-    @test_approx_eq pfftd2_m4[i] true_fftd2_m4[i]
-    @test_approx_eq pifft_fft_m4[i] m4[i]
-    @test_approx_eq pfftn_m4[i] true_fftn_m4[i]
-    @test_approx_eq pifftn_fftn_m4[i] m4[i]
-
-    @test_approx_eq pfft!_m4[i] true_fft_m4[i]
-    @test_approx_eq pfft!d2_m4[i] true_fftd2_m4[i]
-    @test_approx_eq pifft!_fft_m4[i] m4[i]
-    @test_approx_eq pfft!n_m4[i] true_fftn_m4[i]
-    @test_approx_eq pifft!n_fftn_m4[i] m4[i]
-
-    @test_approx_eq sfftn_m4[i] true_fftn_m4[i]
-    @test_approx_eq sfft!n_m4[i] true_fftn_m4[i]
-    @test_approx_eq psfftn_m4[i] true_fftn_m4[i]
-    @test_approx_eq psfft!n_m4[i] true_fftn_m4[i]
+b = rand(17,14)
+b[3:6,9:12] = m4
+sm4 = slice(b,3:6,9:12)
+
+m3d = map(Float32,reshape(1:5*3*2, 5, 3, 2))
+true_fftd3_m3d = Array(Float32, 5, 3, 2)
+true_fftd3_m3d[:,:,1] = 17:2:45
+true_fftd3_m3d[:,:,2] = -15
+
+# use invoke to force usage of CTPlan versions even if FFTW is present
+for A in (Array,SubArray)
+    for f in (:fft,:ifft,:plan_fft,:plan_ifft)
+        f_ = symbol(string(f, "_"))
+        @eval begin
+            $f_{T,N}(x::$A{T,N}) = invoke($f, (AbstractArray{T,N},), x)
```

Comment:
This should be `invoke($f, Tuple{AbstractArray{T,N}}, x)` and similarly below. With this change the inference tests pass for me, although there are a few deprecation warnings that need to be fixed.